src/plugins/intel_gpu/src/graph/impls/ocl/ctc_greedy_decoder.cpp

```diff
@@ -31,12 +31,31 @@ struct ctc_greedy_decoder_impl : typed_primitive_impl_ocl<ctc_greedy_decoder> {
         auto has_second_output = !primitive->second_output.empty();
         params.inputs.push_back(convert_data_tensor(impl_param.input_layouts[1]));
         params.merge_repeated = primitive->ctc_merge_repeated;
-        params.blank_index = primitive->blank_index;
-        params.outputs_num = has_second_output ? 2 : 1;
 
-        if (params.outputs_num == 2) {
-            params.inputs.push_back(convert_data_tensor(impl_param.get_input_layout(1)));
+        if (primitive->use_multiple_outputs) {
+            if (primitive->blank_index == UINT32_MAX) {
+                params.blank_index = impl_param.get_input_layout(0).spatial(1) - 1;
+            } else {
+                params.blank_index = primitive->blank_index;
+            }
+            params.outputs_num = 2;
+            params.use_multiple_outputs = primitive->use_multiple_outputs;
```

Comment:
If use_multiple_outputs are only necessary for kernel params, I think the node don't need to have it.

```suggestion
            params.use_multiple_outputs = (primitive->num_outputs != 1);
```

Reply:
use_multiple_outputs in primitive is initialized as false and used to check dynamic shape case. And num_outputs cannot differentiate dynamic shape and  static shape cases. So use_multiple_outputs is needed.